### PR TITLE
Table resize

### PR DESCRIPTION
**Pull request by JoYellow**
added a new option in the table menu to resize and align the table.

### CODE REVIEW

File: document/static/js/modules/schema/document.js [hunk 1]
```diff
@@ -407,15 +407,55 @@ spec.nodes = spec.nodes.update(
...
+                        attrs.width = "75",
+                        attrs.class = "table-75"
+                        break;
+                    default:
+                        attrs.width = "100",
+                        attrs.class = "table-100"
+                        break;
+                }
+                if(attrs.width === "100"){
+                    node.attrs.aligned = 'center'
```
Comment (johanneswilm) [inline document/static/js/modules/schema/document.js]: we shouldn't assign values on the node here. We are meant to translate the node to an HTML entity - not modify it.

File: document/static/js/modules/editor/menus/headerbar/model.js [hunk 2]
```diff
@@ -803,7 +804,28 @@ export const headerbarModel = () => ({
                         deleteTable(editor.currentView.state, editor.currentView.dispatch)
                     },
                     disabled: editor => !findTable(editor.currentView.state)
-                }
+                },
+                {
+                    title: gettext('Resize/Reposition'),
+                    type: 'action',
+                    tooltip: gettext('Insert a table into the document.'),
```
Comment (johanneswilm) [inline document/static/js/modules/editor/menus/headerbar/model.js]: Is "Insert a table into the document." really the best tooltip description for a resize/reposition dialog?

File: document/static/js/modules/schema/document.js [hunk 3]
```diff
@@ -407,15 +407,55 @@ spec.nodes = spec.nodes.update(
...
+                    track: parseTracks(dom.dataset.track),
+                    width: dom.width,
+                    aligned: dom.aligned
                 }
             }}],
             toDOM(node) {
-                const attrs = {}
+                const attrs ={}
+                switch(node.attrs.width){
+                    case "25":
```
Comment (johanneswilm) [inline document/static/js/modules/schema/document.js]: Why make this long switch? It seems like something like this should cover it:

```
attrs['data-width'] = node.attrs.width
attrs['class'] = `table-${node.attrs.width}`

File: document/static/js/modules/schema/document.js [hunk 4]
```diff
@@ -407,15 +407,55 @@ spec.nodes = spec.nodes.update(
...
+                        break;
+                    default:
+                        attrs.width = "100",
+                        attrs.class = "table-100"
+                        break;
+                }
+                if(attrs.width === "100"){
+                    node.attrs.aligned = 'center'
+                }
+                switch(node.attrs.aligned){
```
Comment (johanneswilm) [inline document/static/js/modules/schema/document.js]: same comment as previous switch statement

**Comment by johanneswilm** [comment]
@rahulalientt You need to sign the CLA for me to be able to merge this.

File: document/static/js/modules/editor/dialogs/table.js [hunk 5]
```diff
@@ -96,3 +97,107 @@ export class TableDialog {
...
+        for (let d = $head.depth; d > 0; d--) if ($head.node(d).type.spec.tableRole == "table") return $head.node(d)
+        return false
+    }
+
+    submitForm(){
+        let table = this.findTable(this.editor.currentView.state)
+        table.attrs.width = this.width
+        table.attrs.aligned = this.aligned
+        deleteTable(this.editor.currentView.state, this.editor.currentView.dispatch)
+        const schema = this.editor.currentView.state.schema
```
[the remark this answers is not in the document]
Comment (rahulalientt) [reply document/static/js/modules/editor/dialogs/table.js]: No.
Removed that code

File: document/static/js/modules/editor/menus/headerbar/model.js [hunk 6]
```diff
@@ -803,7 +804,28 @@ export const headerbarModel = () => ({
...
+                    action: editor => {
+                        // let table = findTable(editor.currentView.state)
+                        // table.attrs.width ="50"
+                        // deleteTable(editor.currentView.state, editor.currentView.dispatch)
+                        // const schema = editor.currentView.state.schema
+                        // editor.currentView.dispatch(
+                        //     editor.currentView.state.tr.replaceSelectionWith(
+                        //         table
+                        //     )
+                        // )
```
[the remark this answers is not in the document]
Comment (rahulalientt) [reply document/static/js/modules/editor/menus/headerbar/model.js]: No
Removed those commented lines

File: document/static/js/modules/schema/document.js [hunk 7]
```diff
@@ -407,15 +407,55 @@ spec.nodes = spec.nodes.update(
...
-                track: {default: []}
+                track: {default: []},
+                width:{default:"100"},
+                aligned:{default:'center'}
             },
             parseDOM: [{tag: "table", getAttrs(dom) {
                 return {
-                    track: parseTracks(dom.dataset.track)
+                    track: parseTracks(dom.dataset.track),
+                    width: dom.width,
```
[the remark this answers is not in the document]
Comment (rahulalientt) [reply document/static/js/modules/schema/document.js]: Fixed in new commit

File: document/static/js/modules/schema/document.js (hunk 1 above)
Comment (rahulalientt) [reply document/static/js/modules/schema/document.js]: Fixed in new commit

File: document/static/js/modules/editor/menus/headerbar/model.js (hunk 2 above)
Comment (rahulalientt) [reply document/static/js/modules/editor/menus/headerbar/model.js]: Changed the tooltip description

File: document/static/js/modules/schema/document.js (hunk 3 above)
Comment (rahulalientt) [reply document/static/js/modules/schema/document.js]: Fixed in new commit

File: document/static/js/modules/editor/dialogs/table.js [hunk 8]
```diff
@@ -134,9 +141,12 @@ export class TableResizeDialog {
     submitForm(){
         let table = this.findTable(this.editor.currentView.state)
         table.attrs.width = this.width
-        table.attrs.aligned = this.aligned
+        table.attrs['data-width'] = this.width
+        this.width == "100" ? table.attrs['data-aligned'] = "center" : table.attrs['data-aligned'] = this.aligned
```
[the remark this answers is not in the document]
Comment (rahulalientt) [reply document/static/js/modules/editor/dialogs/table.js]: Fixed in new commit

**Pull request merged**